server/protocols/webadmin/webadmin-cassandra-data/src/main/java/org/apache/james/webadmin/service/CassandraMappingsSolveInconsistenciesTask.java

```diff
@@ -0,0 +1,56 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import javax.inject.Inject;
+
+import org.apache.james.rrt.cassandra.CassandraMappingsSourcesDAO;
+import org.apache.james.rrt.cassandra.migration.MappingsSourcesMigration;
+import org.apache.james.task.Task;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraMappingsSolveInconsistenciesTask implements Task {
+    public static final String TYPE = "cassandraMappingsSolveInconsistencies";
+
+    private final MappingsSourcesMigration mappingsSourcesMigration;
+    private final CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO;
+
+    @Inject
+    CassandraMappingsSolveInconsistenciesTask(MappingsSourcesMigration mappingsSourcesMigration,
+                                              CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO) {
+        this.mappingsSourcesMigration = mappingsSourcesMigration;
+        this.cassandraMappingsSourcesDAO = cassandraMappingsSourcesDAO;
+    }
+
+    @Override
+    public Result run() {
+        return cassandraMappingsSourcesDAO.truncateTable()
+            .doOnError(e -> LOGGER.error("Error while cleaning up data in mappings sources projection table"))
+            .then(Mono.fromCallable(mappingsSourcesMigration::run))
+            .onErrorResume(e -> Mono.just(Result.PARTIAL))
```

Comment:
you don't log that error?

Reply:
what if `Mono.fromCallable(mappingsSourcesMigration::run)` returns an error Mono then?